### ISSUE TITLE
[HOLD for payment 2022-09-23] [$250] Web - Login - When the page size is reduced, the information on the banner is cut off

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#10410]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to Login page NewDot 
2. Decrease the screen width

## Expected Result:
When the page size is reduced, the information on the banner is cut off.
Clipped only beige banner.

## Actual Result:
Describe what actually happened

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.88.13

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/185000492-0fe8cfef-32d5-4728-b78c-a8bd6ac59b02.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kakajann** [comment]
# Proposal
This line:
https://github.com/Expensify/App/blob/25a7a90c63d3b1d9c8d321f48323c1b1c0d3e9a4/src/pages/signin/SignInPageLayout/index.js#L56
should be: 
```jsx
resizeMode="contain"
```

# Result
| Before | After |
| - | - |
| https://user-images.githubusercontent.com/11542415/185454489-3f18bdd2-ac88-4037-b130-27466e9658b8.mov | https://user-images.githubusercontent.com/11542415/185455101-182109fc-00c4-4e43-a105-97652f532f28.mov |

**Comment by thesahindia** [comment]
@Beamanator, we can close this one as we already have an internal [issue](https://github.com/Expensify/Expensify/issues/213459) for this.

**Comment by Beamanator** [comment]
@thesahindia good find!

@ctkochan22 any reason you'd prefer making this issue internal? If this is just a styling thing let's make it external, but if there's other considerations please let me know :D

**Comment by ctkochan22** [comment]
Lets handle this [issue](https://github.com/Expensify/Expensify/issues/213459) here.

**Comment by ctkochan22** [comment]
I don't think just switching it to contain works. We use `"cover"` so that when the area is larger than the image, it scales up accordingly. If we just applied "contain" it would not do so.

@shawnborton I introduced this bug originally here: https://github.com/Expensify/App/pull/7771 (sorry). I think the issue is that the graphic/text for the CPA image is closer to the margin than the usual background images. 
<img width="686" alt="image" src="https://user-images.githubusercontent.com/12980144/186521751-a34f3bbe-a89b-4634-a6dc-32a8058eba0e.png">
- We are pulling the images here: https://github.com/Expensify/App/blob/a70aa0cd498820662bd9ac4138026095e3e0ed31/src/styles/StyleUtils.js#L379
- When we reach a certain screen width, `isMediumScreenWidth` we switch the `<Image>` component's "resizeMode" from `"cover"` to `"contain"`: https://github.com/Expensify/App/blob/a0c8daf2fb9d59d9db903b8cf86ae72bb9f1d3af/src/pages/signin/SignInPageLayout/index.js#L56

I think the issue is that `isMediumScreenWidth` turns true after we cut off the CPA banner.

**Comment by arielgreen** [comment]
Unassigning myself since we're fixing internally.

**Comment by shawnborton** [comment]
What happens if we switch contain to cover in the `isMediumScreenWidth` scenario?

I would be down to just do nothing here.

**Comment by ctkochan22** [comment]
So "cover" will scale it up, the larger the window gets it'll scale up. While "contain" will always make sure the image is shown by scaling down. 

Here is a video: https://recordit.co/kWeNVNLzUo

**Comment by ctkochan22** [comment]
So if we just set it as "contain", it wont' scale up per se. So stretched out , it'll look like this: 
<img width="800" alt="image" src="https://user-images.githubusercontent.com/12980144/186548131-bb5e9867-a561-4cc6-9385-093dfe3436ad.png">
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/12980144/186548162-fbfa1789-996d-4979-94a4-75ebf1707752.png">


**Instead of:**
<img width="1130" alt="image" src="https://user-images.githubusercontent.com/12980144/186548196-e438d503-9786-4792-ae0a-81ac3941e474.png">
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/12980144/186548215-f7eef580-05cd-4e32-a696-9bbd56cc3aed.png">

**Comment by shawnborton** [comment]
But what about keeping it as contain just at the IsMedium width? Maybe that will just force the whole thing to be shown at that very particular width?

**Comment by ctkochan22** [comment]
> But what about keeping it as contain just at the IsMedium width?

I think we are already doing that, https://github.com/Expensify/App/blob/a0c8daf2fb9d59d9db903b8cf86ae72bb9f1d3af/src/pages/signin/SignInPageLayout/index.js#L56

At least if I'm understanding you correctly. So if isMedium width is true, use "contain". But for anything greater, use "cover"

**Comment by shawnborton** [comment]
Maybe I am suggesting the opposite then. What happens when we just use cover for isMedium?

**Comment by kakajann** [comment]
`cover` is also responsive. you can check it in my proposal video

**Comment by shawnborton** [comment]
Okay so that might do the trick then at the isMedium size - can you also share what it looks like for the CPA card image?

**Comment by kakajann** [comment]
## with CPA card image

https://user-images.githubusercontent.com/11542415/187011018-ce99a8b8-4691-4421-8d91-0f8fafba95b7.mov

**Comment by shawnborton** [comment]
Okay I think it's fine like that? Maybe a side by side of IsMedium and IsLarge breakpoints could be helpful too, but I think what I see above in the video would be totally fine.

**Comment by ctkochan22** [comment]
Yeah, all for the simplest solution. `contain` is responsive in that it will get scale smaller, and will enlargen up until the size of the image. It won't scale up if the window is larger.

**Comment by kakajann** [comment]
Considering the image is `svg`, I think its enlargement will depend on the parents' width and height

**Comment by shawnborton** [comment]
Hmm but can't we only use contain at the IsMedium breakpoint? And then after that, once we hit IsLarge, we use cover?

**Comment by shawnborton** [comment]
I apologize, I am finding it a bit hard to follow how this is working exactly. In the video above, how did we achieve this:
<img width="255" alt="image" src="https://user-images.githubusercontent.com/2319350/187955065-a249ebf8-9353-42b3-9a71-0c205e610e1f.png">


That seems like it will solve the problem, no?

**Comment by kakajann** [comment]
> In the video above, how did we achieve this

My making resize mode `contain`. and IMO it solves this issue.

**Comment by shawnborton** [comment]
And sounds like we can do that for just this one particular image right?

**Comment by kakajann** [comment]
No, it applies to all images

**Comment by shawnborton** [comment]
Got it, I think as long as we just take a good look at what contain will do to the other image style at some standard sizes (desktop at say 1400x800, etc) then we'd be good to go with that solution.

**Comment by kakajann** [comment]
ResizeMode = `contain`
Resolution = `1400x800`
| <img width="1014" alt="Screen Shot 2022-09-03 at 3 04 42 AM" src="https://user-images.githubusercontent.com/11542415/188241063-450d8b7a-1a62-4aef-9b0f-74cc8cd86df2.png"> | <img width="1014" alt="Screen Shot 2022-09-03 at 3 04 51 AM" src="https://user-images.githubusercontent.com/11542415/188241096-c3b6ee7f-a80a-4a10-abe2-b0064a561554.png"> |
| - | - |
| <img width="1014" alt="Screen Shot 2022-09-03 at 3 04 15 AM" src="https://user-images.githubusercontent.com/11542415/188241028-43bc63c8-836f-44b1-b690-4dc30e8b947b.png"> | <img width="1014" alt="Screen Shot 2022-09-03 at 3 04 08 AM" src="https://user-images.githubusercontent.com/11542415/188240990-f7bd255f-4835-4b85-8cad-3f03ad466caa.png"> |

**Comment by shawnborton** [comment]
Cool, that looks pretty solid to me. I don't think we need to invest too much time on this one too as I would imagine we'll change this up in the coming months.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by ctkochan22** [comment]
So long as its `contain` the full image will be visible, but it wont expand if the resolution is large. 

I also think this isn't worth too much effort. Lets switch it to only `contain` and go from there. If we don't like it, we can address it later

**Comment by ctkochan22** [comment]
@dylanexpensify can we get a upwork job created and give it to @kakajann . Should be a super easy change and quick reivew/merge

**Comment by dylanexpensify** [comment]
internal: https://www.upwork.com/ab/applicants/1567103025183875072/job-details
external: https://www.upwork.com/jobs/~0140cd4b0b98c2f40a

@kakajann I also invited you to apply!

**Comment by dylanexpensify** [comment]
@ctkochan22 should I put `exported` and grab a C+? Or just get @kakajann paid out?

**Comment by ctkochan22** [comment]
Slap an exported label and I'll approve the proposal

**Comment by dylanexpensify** [comment]
offer sent @kakajann!

**Comment by dylanexpensify** [comment]
Payment sent, contract closed, posting removed!

**Comment by mananjadhav** [comment]
@dylanexpensify A bit confused here, is C+ required here? and PR is not merged yet it says Payment sent. 

Quick note, I went through the history, I'll review the PR today in a few hours(if it still needs a review).

**Comment by dylanexpensify** [comment]
@mananjadhav ah this is my fault, I thought we needed to create/assign the job and pay it out, but I see we needed to wait for review and merge. Please proceed as normal with review when you can! cc @ctkochan22

**Comment by mananjadhav** [comment]
Thanks for clarifying. I'll get to it in sometime.

**Comment by dylanexpensify** [comment]
Any update @mananjadhav?

**Comment by mananjadhav** [comment]
@dylanexpensify PR is merged and deployed to staging 2 days back

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by dylanexpensify** [comment]
reassigning for payment as I'm OOO tomorrow!

**Comment by puneetlath** [comment]
Hm, it looks like @kakajann was already paid for this job.

@mananjadhav sent you a hiring offer for C+.

**Comment by mananjadhav** [comment]
Accepted @puneetlath

**Comment by puneetlath** [comment]
All paid!
